addons/account/models/account_account.py

```diff
@@ -597,37 +614,34 @@ def _compute_opening_debit_credit(self):
     @api.depends('code')
     def _compute_account_type(self):
         accounts_to_process = self.filtered(lambda account: account.code and not account.account_type)
-        self._get_closest_parent_account(accounts_to_process, 'account_type', default_value='asset_current')
+
+        if accounts_to_process:
+            company_accounts = self.search(self._check_company_domain(self.env.company))
```

Comment:
In this case, since this compute is triggered by a change in `code` (which is changed for the currently active company) and we effectively only want the compute to modify the value once, I think it's okay